xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/js/xwiki/usersandgroups/usersandgroups.js

```diff
@@ -397,8 +397,11 @@ function displayUsersAndGroups(row, i, table, idx, form_token, targetDocument)
 {
   var userurl = row.userurl;
   var uorg = table.json.uorg;
-  var allows = row.allows;
-  var denys = row.denys;
+  // Rights are case insensitive in XWiki security module, and the reference rights defined in rightsUI.vm
+  // are lowercase, hence we make sure to convert all applying rights to lowercase before comparing them with the
+  // reference ones.
+  var allows = row.allows.map(entry => entry.toLowerCase());
+  var denys = row.denys.map(entry => entry.toLowerCase());
```

Comment:
FWIW `toLowerCase()` should be avoided. The signature taking a locale should be used. Here it's fine because we're pretty sure that the right will not contain some characters. However it's best to use `Locale.ROOT`. See https://rules.sonarsource.com/java/RSPEC-1449

Reply:
EDIT: I thought this was java... I have no idea about javascript, needs to be researched....

Reply:
Seems the issue is the same: https://www.w3schools.com/jsref/jsref_tolocalelowercase.asp But I don't know what's the recommended solution.

Reply:
So it seems that in javascript `toLowerCase()` is the method to use when case insensitive and `toLocaleLowerCase()` is locale sensitive. So it looks good here. We should review and fix our java usages though.

Reply:
> We should review and fix our java usages though.

For ex: https://github.com/xwiki/xwiki-platform/blob/bc6fb452ef10414c1756b2b547e8129ec1e35c9d/xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/uorgsuggest.vm#L22